Sources/MapLibreSwiftUI/MapView.swift

```diff
@@ -9,6 +9,7 @@ public struct MapView: UIViewRepresentable {
 
     let styleSource: MapStyleSource
     let userLayers: [StyleLayerDefinition]
+    var mapViewModifier: ((MLNMapView) -> Void)?
```

Comment:
In general, views in SwiftUI shouldn't contain mutable state outside the provided state frameworks.

I think this works for basic cases as-is, but I'm already getting a headache trying to sort out whether there is a failure mode in which an update isn't picked up.

Is it possible for your use case to make this `let` instead and add it to the constructor with a default value (`nil`)? You could probably also go one step further for ergonomics and write a modifier that returns a new `MapView` with this set to a new value (probably easiest to make a private "copy constructor" that takes the raw values rather than builders to facilitate)?

```suggestion
    let mapViewModifier: ((MLNMapView) -> Void)?
```

Reply:
So my thought here was that as this isn't defined as `public`, the only way to modify this is from outside is via the defined modifier - as this is all a Struct I do not see how an update could not be picked up? By making it a var we let swift handle the copying for us if needed instead of having to maintain the private copy constructor and modifier every time we modify the parameters of MapView. Instead I would rename the modifier to unsafeMapViewModifier and add documentation to explain that this is an escape hatch for now and "here be dragons".

Alternatively we not make this a modifier at all, but just part of `init` as you suggest with a default of nil - I'm personally just not a fan of double closures - but they are very common in SwiftUI, like for `Button`, so in this case I guess that would be fine too.

I will update the PR to unsafeMapViewModifier first so we see how it feels, but happy to change it to a non modifier if it doesn't feel good enough.

Reply:
Ignore me; that was a dum comment. I think you're correct.